processing/src/main/java/org/apache/druid/data/input/impl/HttpEntity.java

```diff
@@ -80,10 +85,15 @@ public Predicate<Throwable> getRetryCondition()
     return t -> t instanceof IOException;
   }
 
-  public static InputStream openInputStream(URI object, String userName, PasswordProvider passwordProvider, long offset)
+  public static InputStream openInputStream(URI object, String userName, PasswordProvider passwordProvider, long offset, final Map<String, String> requestHeaders)
       throws IOException
   {
     final URLConnection urlConnection = object.toURL().openConnection();
+    if (requestHeaders.size() > 0) {
```

Comment:
also need to check that requestHeaders is not null.

Reply:
if not, then requestHeaders is not nullable.

Reply:
fixed